### DISCUSSION TITLE
Smartmeter: publish MQTT when value changed more then xy

### DISCUSSION THREAD

**Discussion by Bubbleman75** [arendst/Tasmota#13313] [Q&A] [answered]
Hey community,

I need your help.
I use a Wemos D1 Mini to readout my smartmeter. Everything works fine.
At the moment the Wemos is transmiting the (current consumption) data immediately/at every change (even +/- 1 Watt). That causes a lot of unnecessary traffic and logging
My target is, not sending on given teleperiod MQTT, but only when the sensor change the value more then e.g. 10 Watt.
How can I realise this?

the current script looks like this:

`>D`
`>B`
`=>sensor53 r`
`>M 1`
`+1,2,o,0,9600,LK11BE`

`1,1-0:1.8.0*255(@1,Gesamtverbrauch,kWh,total,4`
`1,1-0:1.8.0*96(@1,Verbrauch 1 Tag,kWh,total_1d,4`
`1,1-0:1.8.0*97(@1,Verbrauch 7 Tage,kWh,total_7d,4`
`1,1-0:1.8.0*98(@1,Verbrauch 30 Tage,kWh,total_30d,4`
`1,1-0:1.8.0*99(@1,Verbrauch 365 Tage,kWh,total_365d,4`
`1,1-0:16.7.0*255(@1,Verbrauch aktuell,W,current,20`
`#`

THX
Thomas

**Comment by barbudor** [comment]
If your smart meter use Tasmota energy, then look at PowerDelta command

**Comment by Bubbleman75** [comment, marked as answer]
I am using the "Smart Meter Interface" feature.
I used online tasmocompiler to compile a tasmota image with the needed SMI feature.
There are no rules, only script.

**Reply by barbudor** [reply]
How does that prevent you to read the doc for PowerDelta?

**Reply by Bubbleman75** [reply]
PowerDelta is a command used in rules, right?
When using the SMI feature rules are deactivated. You only can use scripting.
[https://tasmota.github.io/docs/Smart-Meter-Interface/](url)
I use a IR-diode to read out the data from my smart meter.

![grafik](https://user-images.githubusercontent.com/5702279/136694285-45925f08-0a51-4ac7-b4e6-ddb463d55858.png)

**Reply by sfromis** [reply]
PowerDelta is not related to rules, it is just a command saving a setting related to energy monitoring reporting.

**Reply by Bubbleman75** [reply]
this is helpful information!

**Reply by barbudor** [reply]
From the beginning....
I never used the word "Rules" anywhere but you know what, ¯\_(ツ)_/¯

**Reply by gemu2015** [reply]
SML driver does not use power delta. 
to solve this you have to handle all computing in the script

no complete script here only fragments to declare what to do:

#define USE_SML_SCRIPT_CMD in user config


1. you have to disable MQTT in SML 
by setting MQTT code to *
+1,2,o,0,9600,*


\>D
lastsml=0

2. then read out energy values and compare and then send individual MQTT messages by script
in the >J
section create a subroutine call by
\>J
%=#jsub

\>S

; send json subroutine 
#jsub
; get from descriptor var 1 (total)
temp=abs(lastsml-sml[1])
if temp>10
then
; response append
rapp MQTT message
endif

**Reply by barbudor** [reply]
@gemu2015 , Isn't SML using the Tasmota Energy model ? Because PowerDelta is managed by the base driver xsns_03_energy, so it should work with any driver that fills Tasmota Energy

**Reply by gemu2015** [reply]
no SML does not use the energy manager!
